contracts/farm/LiquidityMiningV1.sol

```diff
@@ -0,0 +1,767 @@
+		_;
+	}
+
+	modifier onlyBeforeMigrationGracePeriodFinished() {
+		require(migrationGracePeriodState < MigrationGracePeriodStates.Finished, "Forbidden: contract deprecated");
+		_;
+	}
+
+	modifier onlyAfterMigrationFinished() {
+		require(migrationGracePeriodState == MigrationGracePeriodStates.Finished, "Forbidden: migration is not over yet");
+		_;
+	}
+
+	/* Functions */
+
+	/**
+	 * @notice Initialize mining.
+	 *
+	 * @param _liquidityMiningV2 The LiquidityMiningV2 contract address
+	 */
+	function initialize(address _liquidityMiningV2) external onlyAuthorized {
+		/// @dev Non-idempotent function. Must be called just once.
+		require(_liquidityMiningV2 != address(0), "Invalid address");
+		liquidityMiningV2 = _liquidityMiningV2;
+	}
+
+	/**
+	 * @notice Sets lockedSOV contract.
+	 * @param _lockedSOV The contract instance address of the lockedSOV vault.
+	 */
+	function setLockedSOV(ILockedSOV _lockedSOV) external onlyAuthorized {
+		require(address(_lockedSOV) != address(0), "Invalid lockedSOV Address.");
+		lockedSOV = _lockedSOV;
+	}
+
+	/**
+	 * @notice Sets unlocked immediately percent.
+	 * @param _unlockedImmediatelyPercent The % which determines how much will be unlocked immediately.
+	 * @dev @dev 10000 is 100%
+	 */
+	function setUnlockedImmediatelyPercent(uint256 _unlockedImmediatelyPercent) external onlyAuthorized {
+		require(_unlockedImmediatelyPercent < 10000, "Unlocked immediately percent has to be less than 10000.");
+		unlockedImmediatelyPercent = _unlockedImmediatelyPercent;
+	}
+
+	/**
+	 * @notice sets wrapper proxy contract
+	 * @dev can be set to zero address to remove wrapper
+	 */
+	function setWrapper(address _wrapper) external onlyAuthorized {
+		wrapper = _wrapper;
+	}
+
+	/**
+	 * @notice stops mining by setting end block
+	 */
+	function stopMining() public onlyAuthorized {
+		require(endBlock == 0, "Already stopped");
+
+		endBlock = block.number;
+	}
+
+	// TODO: this should only be used by the LiquidityMiningV2 contract??
+	/// @notice This function starts the migration process which involves two steps:
+	///         1. Starts the migration grace period when people can withdraw or claim for rewards
+	///					2. Stops mining, i.e., no more rewards are paid
+	function startMigrationGracePeriod() external onlyAuthorized onlyBeforeMigrationGracePeriod {
+		migrationGracePeriodState = MigrationGracePeriodStates.Started;
+		stopMining();
+	}
+
+	// TODO: this should only be used by the LiquidityMiningV2 contract??
+	/// @notice This function finishes the migration process disabling further withdrawals and claims
+	/// @dev migration grace period should have started before this function is called.
+	function finishMigrationGracePeriod() external onlyAuthorized onlyBeforeMigrationGracePeriodFinished {
+		require(migrationGracePeriodState == MigrationGracePeriodStates.Started, "Migration hasn't started yet");
```

Comment:
The `require` introduces a harder contraint making the `onlyBeforeMigrationGracePeriodFinished` worthless ... as suggested by Haku we might create a new modifier `onlyMigrationGracePerioidStarted` that checks the same condition as the `require`

Reply:
👍